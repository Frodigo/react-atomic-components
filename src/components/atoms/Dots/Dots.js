import React from 'react';
import PropTypes from 'prop-types';

import classes from './Dots.module.css';

/**
 * This are the dots.
 * @param {*} props
 */
const Dots = (props) => {
    const { activeItemId, items, onClick, ...rest } = props;

    const children = items.map((dot) => {
        const isActive = dot.id === activeItemId;

        return (
            <button
                type="button"
                onClick={onClick(dot)}
                key={dot.id}
                className={isActive ? classes.dotActive : classes.dot}
            >
                {dot.label}
            </button>
        );
    });

    return (
        <nav className={classes.dots} {...rest}>
            {children}
        </nav>
    );
};

Dots.propTypes = {
    activeItemId: PropTypes.number,
    items: PropTypes.arrayOf(
        PropTypes.shape({
            id: PropTypes.number.isRequired,
            label: PropTypes.string.isRequired
        })
    ).isRequired,
    onClick: PropTypes.func.isRequired
};

export default Dots;
