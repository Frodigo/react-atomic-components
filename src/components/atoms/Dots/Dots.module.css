.dots {
    align-items: center;
    display: flex;
    justify-content: center;
    margin: var(--spacing-xs) 0;
}

.dot {
    background: none;
    border: solid 1px var(--color-tiertary);
    border-radius: 50%;
    cursor: pointer;
    height: 20px;
    font-size: 0;
    margin: 0 5px;
    outline: none;
    width: 20px;
}

.dotActive {
    composes: dot;
    background-color: var(--color-tiertary);
    cursor: default;
    pointer-events: none;
}
