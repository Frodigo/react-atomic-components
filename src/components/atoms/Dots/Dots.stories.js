import React from 'react';

import { default as DotsComponent } from './Dots';

export default {
    title: 'Atoms/Controls/Dots',
    component: DotsComponent,
    argTypes: {
        onClick: { action: 'clicked' }
    }
};

const Template = (args) => <DotsComponent {...args} />;

export const Dots = Template.bind({});
Dots.args = {
    items: [
        {
            id: 1,
            label: '1'
        },
        {
            id: 2,
            label: '2'
        },
        {
            id: 3,
            label: '3'
        },
        {
            id: 4,
            label: '4'
        }
    ],
    activeItemId: 2
};
