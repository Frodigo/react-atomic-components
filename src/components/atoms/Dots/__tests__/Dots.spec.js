import React from 'react';
import { render } from '@testing-library/react';

import Dots from '../Dots';

describe('Dots should be rendered', () => {
    const onCLickHandler = jest.fn();
    const props = {
        items: [
            {
                id: 1,
                label: '1'
            },
            {
                id: 2,
                label: '2'
            },
            {
                id: 3,
                label: '3'
            },
            {
                id: 4,
                label: '4'
            }
        ],
        activeItemId: 2
    };

    test('Dots renders correctly', () => {
        const { container } = render(
            <Dots {...props} onClick={onCLickHandler} />
        );

        expect(container).toMatchSnapshot();
    });

    test('Active item should be marked', () => {
        const { container } = render(
            <Dots {...props} onClick={onCLickHandler} />
        );

        const elements = container.firstChild.children;

        expect(elements.item(1).classList.contains('dotActive')).toBe(true);
    });
});
