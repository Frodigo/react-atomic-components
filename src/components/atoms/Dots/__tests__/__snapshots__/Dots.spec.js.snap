// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dots should be rendered Dots renders correctly 1`] = `
<div>
  <nav
    class="dots"
  >
    <button
      class="dot"
      type="button"
    >
      1
    </button>
    <button
      class="dotActive"
      type="button"
    >
      2
    </button>
    <button
      class="dot"
      type="button"
    >
      3
    </button>
    <button
      class="dot"
      type="button"
    >
      4
    </button>
  </nav>
</div>
`;
